### ISSUE TITLE
Classic Sunday: Don't suggest intermediate fix that prompts another fix

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15131]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/NumericPredicate + Style/BitwisePredicate

### Steps to reproduce the problem

```ruby
stat = File.stat('.')
pp stat.mode & 0o077 == 0o000
# Use `(stat.mode & 0o077).zero?` instead of `stat.mode & 0o077 == 0o000`. [[Correctable] Style/NumericPredicate]
pp (stat.mode & 0o077).zero?
# Replace with `stat.mode.nobits?(0o077)` for comparison with bit flags. [[Correctable] Style/BitwisePredicate]

```

### Expected behavior

Suggest the second cop in the first place

### Actual behavior

Gives coder community service first, only to find out he should have gone straight to jail.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
2026-04-26 09:09:37.074 system_profiler[9689:10477311] hw.cpufamily: 0x1b588bb3
System Version: macOS 26.3.1 (a) (25D771280a)
Model Identifier: MacBookAir10,1
ruby 4.0.3 (2026-04-21 revision 85ddef263a) +PRISM [arm64-darwin25]
rubocop     2026-04-25 80e48eaa8
```

**Comment by rafa-el-souza** [comment]
Hi! I've investigated the issue on the current master branch, and the cause is that every pattern in `Style/BitwisePredicate` needs the bit operation to be parenthesized (a `begin` node):

https://github.com/rubocop/rubocop/blob/84bcf77f7ce3d79586c8ae540412bbc9b50aa27e/lib/rubocop/cop/style/bitwise_predicate.rb#L67-L71

That turns `stat.mode & 0o077 == 0o000` into a false negative for it. The cop that does match is `Style/NumericPredicate`, its autocorrect adds the parens, and only then `BitwisePredicate` matches.

I'd propose to make `BitwisePredicate`'s patterns also match unparenthesized comparisons (`x & y == 0`, `!= 0`, `> 0`, `>= 1`, and `x & y == y` for `allbits?`). Since predicate methods bind tighter than `&`, they are unaffected.

But that would mean `x & y == 0` would be flagged by both cops. Would this be acceptable or should `Style/NumericPredicate` not flag when the receiver is a bitwise operation and `Style/BitwisePredicate` is enabled (via `config.cop_enabled?`)? Happy to implement either way.

**Comment by forthrin** [comment]
Assuming preventing double or sequential cops is a recurring issue and devs have ways of dealing with them.

**Comment by forthrin** [comment]
@rafa-el-souza: Did you decide on an approach?
